api/v1beta1/solrcloud_types.go

```diff
@@ -1022,3 +1044,35 @@ type SolrTLSOptions struct {
 	// +optional
 	RestartOnTLSSecretUpdate bool `json:"restartOnTLSSecretUpdate,omitempty"`
 }
+
+type SolrSecurityOptions struct {
+	// Secret containing credentials the operator should use for API requests to secure Solr pods.
+	// If you provide this secret, then the operator assumes you've also configured your own security.json file and
+	// uploaded it to Solr. The 'key' of the secret selector is the username. If you change the password for this
+	// user using the Solr security API, then you *must* update the secret with the new password or the operator will be
+	// locked out of Solr and API requests will fail, ultimately causing a CrashBackoffLoop for all pods if probe endpoints
+	// are secured.
+	//
+	// If you don't supply this secret, then the operator bootstraps a default security.json file and creates a
+	// corresponding secret containing the credentials for three users: admin, solr, and k8s-oper. All API requests
+	// from the operator are made as the 'k8s-oper' user, which is configured with minimal access. The 'solr' user has
+	// basic read access to Solr resources. Once the security.json is bootstrapped, the operator will not update it!
+	// You're expected to use the 'admin' user to access the Security API to make further changes. It's strictly a
+	// bootstrapping operation.
+	// +optional
+	BasicAuthSecret *corev1.SecretKeySelector `json:"basicAuthSecret,omitempty"`
```

Comment:
I am not familiar with how other CRDs handle this, but should the "key" have to be the username, or should it be passed in separately?

Like `basicAuthUsername` and `basicAuthPasswordSecret`. 

[Looking at the kubernetes docs](https://kubernetes.io/docs/concepts/configuration/secret/#basic-authentication-secret), it looks like the standard for Basic Auth secrets is to store "username": <> and "password": <>.

Reply:
We're storing creds for multiple users: admin, solr, and k8s-oper as well as the security.json (so we don't have to recompute it for every reconcile loop). So our secret is not a K8s basic auth secret

Reply:
One idea I liking more and more is to have 2 secrets, one that is a `"kubernetes.io/basic-auth"` that holds the creds for the `k8s-oper` user and another `Opaque` that holds the bootstrapped `admin` user and the `security.json`. We can name the latter `<cloud>-solrcloud-security-bootstrap` (or similar) which makes it clear it's only for bootstrapping security and the former holds the creds for the user needed by the operator.